data.md

```diff
@@ -130,6 +130,13 @@ We also add `undefined` as a value, which makes many partial functions in the se
  // --------------------------
 ```
 
+Some values are optional.
+When convenient, they can be represented by wrapping them in the following data types.
+
+```k
+syntax OptionInt ::= "some" Int | "none"
```

Comment:
In K, traditionally we'll do this with subsorting instead of using an injection (which a language without subtyping would require). Would recommend doing this instead, it's more idiomatic.

```k
     syntax MemAddr ::= Int | ".MemAddr"
  // -----------------------------------
```

Note this enables us to check if it's an `Int` or a `MemAddr` with sort predicates (which are automatically generated for each sort, in this case `syntax Bool ::= isMemAddr ( K )` and `syntax Bool ::= isInt ( K )`), while there currently isn't a well-supported way of turning pattern matching into predicates (other than splitting into multiple rules).

Reply:
Resolved by using `Map` instead.